etcd3/client.py

```diff
@@ -86,34 +74,94 @@ def __init__(self, alarm_type, member_id):
         self.member_id = member_id
 
 
+class Node(object):
+    """Represents an etcd cluster member."""
+
+    creds = None
+    time_retry = 300.0
+
+    def __init__(self, host, port, secure=True):
+        self.host = host
+        self.netloc = "{host}:{port}".format(host=host, port=port)
+        self.secure = secure
+        self.protocol = 'https' if secure else 'http'
+        if self.secure and self.creds is None:
+            raise ValueError('Please set credentials for secure connections')
+        self.in_use = False
+        self.last_failed = 0
+        self.channel = self._mkchannel()
+
+    def fail(self):
+        """Set the node to failed."""
+        self.in_use = False
+        self.last_failed = time.time()
+
+    def use(self):
+        """Set the node to in use."""
+        if self.failed:
+            raise ValueError('trying to use a failed node')
+        self.in_use = True
+        self.last_failed = 0
+        return self.channel
+
+    def __str__(self):
+        return "Node({}://{})".format(
+            self.protocol,
+            self.netloc)
+
+    @property
+    def failed(self):
+        return (time.time() - self.last_failed < self.time_retry)
+
+    def _mkchannel(self):
+        if self.secure:
+            return grpc.secure_channel(
+                self.netloc, self.creds)
+        else:
+            return grpc.insecure_channel(self.netloc)
+
+
 class Etcd3Client(object):
-    def __init__(self, host='localhost', port=2379,
-                 ca_cert=None, cert_key=None, cert_cert=None, timeout=None):
-        self._url = '{host}:{port}'.format(host=host, port=port)
+    def __init__(self, host='localhost', port=2379, srv_record=None,
+                 ca_cert=None, cert_key=None, cert_cert=None,
+                 timeout=None, reconnect=False):
```

Comment:
I agree but I couldn't come up with anything better: if you have a better name, I'm happy to use it!